### ISSUE TITLE
memmap behaviour change in master when not specifying shape

### ISSUE THREAD

**Issue by lesteve** [numpy/numpy#11305]
Quickly looking at the recent changes this may be related to https://github.com/numpy/numpy/pull/11223.

#### A snippet to reproduce:
```py
import numpy as np

fname = '/tmp/test.mmap'
with open(fname, 'wb') as f:
    f.write(b'a'*16)

mm = np.memmap(fname, dtype=np.float64)
print(mm.shape)
```

#### Output with numpy 1.14.3:
```
(2,)
```
This is what I would expect: the file contains 16 bytes which fits two float64. This also seems to agree with the `shape` docstring of `np.memmap`: "By default, the returned array will be 1-D with the number of elements determined by file size and data-type."

#### Output with numpy 1.15.0.dev0+c3f9967:
```
()
```

Context: this change is causing failures in the scikit-learn tests using numpy master (this failure was first seen 7 days ago). We can work around this behaviour change if needed, but it does look like this behaviour change did not happen on purpose.

**Comment by pgunn** [comment]
@seberg 
Will do

**Comment by pgunn** [comment]
Will have a PR that should fix the issue within the hour; was a mistake that happened during rewrites in peer review.

**Comment by pgunn** [comment]
This should fix it I think
https://github.com/numpy/numpy/pull/11307

**Comment by charris** [comment]
Fixed by #11307 .
